csharp/test/Ice/objects/AllTests.cs

```diff
@@ -310,31 +310,21 @@ public static Test.InitialPrx allTests(global::Test.TestHelper helper)
                     output.Flush();
                     var top = new Test.Recursive();
                     var p = top;
-                    int depth = 0;
+                    int maxDepth = 100;
                     try
                     {
-                        for (; depth <= 1000; ++depth)
+                        for (int i = 0; i <= maxDepth; i++)
                         {
                             p.v = new Test.Recursive();
                             p = p.v;
-                            if ((depth < 10 && (depth % 10) == 0) ||
-                              (depth < 1000 && (depth % 100) == 0) ||
-                              (depth < 10000 && (depth % 1000) == 0) ||
-                              (depth % 10000) == 0)
-                            {
-                                initial.setRecursive(top);
-                            }
                         }
-                        test(!initial.supportsClassGraphDepthMax());
+                        initial.setRecursive(top);
```

Comment:
Should we actually make two calls:
 - one with max - 1 elements, and verify it succeeds
 - on with max elements, and verify it fails with the expected exception?